### DISCUSSION TITLE
/onedrive vs. ~/OneDrive

### DISCUSSION THREAD

**Discussion by addohm** [abraunegg/onedrive#2485] [Q&A]
Other than the obvious differences listed below, what is the functional difference between these two locations within the container?

/onedrive contains the mapped config and data directories
/OneDrive contains _some_ files; files that are also in the /onedrive/data directory

And to head off the sarcastic fellas like myself  :)
/ is root
~/ is home
onedrive != OneDrive as linux is case sensitive

**Comment by abraunegg** [comment]
@addohm 

Unsure what your question is.

The application will use 2 parameters for various aspects:
* `--confdir` - this is where all the configuration files live. By *default* this is `$HOME/.config/onedrive` .. which should translate to:
  * Normal User: `/home/username/.config/onedrive`
  * Root User: `/root/.config/onedrive
  * **Note:** In some instances, such as cron jobs and/or systemd services, this may even use a configuration directory `/etc/onedrive` however this is *rare* that this is used
  * **Note:** You can change this location to be *anywhere* on your system, so long as you specify this location 100% of the time at application startup and the user that is running the application has permissions to access (read|write) that location
* `--syncdir` or 'sync_dir' config file option - this is where all of your data that is synced between Microsoft OneDrive and your local system will reside. By default, this is `$HOME/OneDrive` which translates to:
  * Normal User: `/home/username/OneDrive`
  * Root User: `/root/OneDrive`
  * **Note:** You can change this location to be *anywhere* on your system, so long as you specify this location 100% of the time at application startup or via configuration file and the user that is running the application has permissions to access (read|write) that location

To validate and view your configuration, use:
```
onedrive --display-config
```

If you are *changing* the configuration directory, you need to specity:
```
onedrive --confdir '/path/to/your/specific/configuration/ --display-config
```

**Reply by addohm** [reply]
Im just wondering what the difference purpose of ~/OneDrive is.  I thought all the files were being stored to the mapped /onedrive/data folder but I noticed the OneDrive folder had some content in it as well.  As a matter of fact, I also found ~/.config/onedrive/ contains all the same files as the mapped /onedrive/config folder.  I'm just trying to understand it and put things together.  It didn't make sense to me that the folders have duplicates.

**Reply by addohm** [reply]
Im still looking for the config file, I'm sure that'll answer most of my questions.

**Reply by abraunegg** [reply]
> Im still looking for the config file, I'm sure that'll answer most of my questions.

The 'config' file does *not* exist by *default* .. you have to create it.

**Reply by addohm** [reply]
No kidding?  I seem to remember a couple years back (before I started dockerizing it) having a config file but everything was commented out.  Guess I'm chasing my tail haha

**Comment by abraunegg** [comment]
> Im just wondering what the difference purpose of ~/OneDrive is. I thought all the files were being stored to the mapped /onedrive/data folder but I noticed the OneDrive folder had some content in it as well. As a matter of fact, I also found ~/.config/onedrive/ contains all the same files as the mapped /onedrive/config folder. I'm just trying to understand it and put things together. It didn't make sense to me that the folders have duplicates.

Your setup sounds *very odd* ... are you using a Docker Container at all ?

In a Docker Container - you *map* `/onedrive/data` as your data location within the Docker Container - but *outside* of the Docker container `onedrive/data` --> `$HOME/OneDrive`

If you have *multiple* Docker images running - you have to have this set up correctly.

**Reply by addohm** [reply]
Im not sure what's odd about it - probably because it's on unraid.  Yes, I'm currently running two onedrive docker containers.  

1) o365 personal
/onedrive/data --> /mnt/user/onedrive/data
/onedrive/config --> /mnt/user/onedrive/config
2) o365 Business
/onedrive/data --> /mnt/user/onedrive-work/data
/onedrive/config --> /mnt/user/onedrive-work/config

My personal account works fine, but its the business account that's been raising all sorts of questions for me.  Not much to do with you as I know your software works.  I've used it for a damn long time.

**Reply by abraunegg** [reply]
OK - so all the paths are 100% OK, internally mapped in the Docker Container and mapped correctly to an external point.

**Reply by addohm** [reply]
So this takes me back to the original question - why am I finding stuff within the containers home folder?  I normally wouldn't be concerned by it but it does bloat the image size unnecessarily.  Are they just psudo-cache?

**Reply by abraunegg** [reply]
> So this takes me back to the original question - why am I finding stuff within the containers home folder? I normally wouldn't be concerned by it but it does bloat the image size unnecessarily. Are they just psudo-cache?

AFAIK there should not be anything *inside* the Docker Container Home Folder ... everything should be mapped externally to external disk .. so nothing should be actually stored *inside* the Docker Container other than what is required to run the 'onedrive' binary.

**Comment by addohm** [comment]
> > So this takes me back to the original question - why am I finding stuff within the containers home folder? I normally wouldn't be concerned by it but it does bloat the image size unnecessarily. Are they just psudo-cache?
> 
> AFAIK there should not be anything _inside_ the Docker Container Home Folder ... everything should be mapped externally to external disk .. so nothing should be actually stored _inside_ the Docker Container other than what is required to run the 'onedrive' binary.

You're not wrong, which is why I came here with my confusions.

Strange...  Just to test, after restarting the container theres nothing in the home folder.  I'll be interested to see what it does when it starts syncing again,.
